### PR TITLE
 Add pending tasks and deploying nodes history

### PR DESCRIPTION
**Pull request by Mykhailenko**
1. Add pending task curve
2. Add deploying curve

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java [hunk 1]
```diff
@@ -201,14 +201,21 @@ private int startTasks(Policy currentPolicy, Map<JobId, JobDescriptor> jobMap, M
             //get rmState and update it in scheduling policy
             Set<String> freeResources = getFreeResources(currentPolicy);
             //if there is no free resources, stop it right now without starting any task
```
Comment (MaelAudren) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: I think that the comment should be moved too.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java [hunk 2]
```diff
@@ -201,14 +201,21 @@ private int startTasks(Policy currentPolicy, Map<JobId, JobDescriptor> jobMap, M
...
-            if (freeResources.isEmpty()) {
-                return 0;
-            }
```
Comment (aminelouati) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: Why did you remove this condition ? 
(you forgot to remove the comment)

Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: Thanks. Indeed i forgot to remote a comment. But condition is below now.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java (hunk 1 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: True

File: common/common-db/src/main/java/org/ow2/proactive/db/RrdDbUtil.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.db;
+
+public class RrdDbUtil {
```
Comment (amouhoub) [inline common/common-db/src/main/java/org/ow2/proactive/db/RrdDbUtil.java]: @Mykhailenko I suggest you rename this class to use the same convention as in:
`log4j.logger.org.ow2.proactive.jmx.RRDDataStore`
maybe `RRDDbUtils` ?

File: rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/rm/RmStatsCommand.java [hunk 4]
```diff
@@ -47,10 +47,9 @@
                                                                      "MaxDeployingNodes", "BusyNodesCount",
                                                                      "AvailableNodesCount", "MaxLostNodes",
                                                                      "MaxFreeNodes", "AverageInactivity", "Status",
-                                                                     "LostNodesCount", "FreeNodesCount",
                                                                      "DownNodesCount", "AverageActivity",
                                                                      "MaxDownNodes", "ConfiguringNodesCount",
-                                                                     "MaxToBeReleasedNodes" };
+                                                                     "MaxToBeReleasedNodes", "NumberPendingTasks" };
```
Comment (amouhoub) [inline rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/rm/RmStatsCommand.java]: Maybe `PendingTasksCount` instead,,, since all other counting metrics are called so ("BusyNodesCount","AvailableNodesCount","DownNodesCount", etc)

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java [hunk 5]
```diff
@@ -119,9 +120,9 @@
                                                     // "ToBeReleasedNodesCo", // useless info
                                                     "BusyNodesCount", //
                                                     "FreeNodesCount", //
-                                                    "DownNodesCount", //
-                                                    "AvailableNodesCount", //
-                                                    "AverageActivity", //
+                                                    "DeployingNodesCount", "DownNodesCount", //
+                                                    "NumberPendingTasks", "AvailableNodesCount", //
```
Comment (amouhoub) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java]: Same as above,,,
btw it's one NumberXXX in the middle of 3 XXXCount

File: rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/ResourceManager.java [hunk 6]
```diff
@@ -557,4 +557,5 @@ NodeSet getNodes(int number, TopologyDescriptor descriptor, List<SelectionScript
      */
     boolean areNodesRecoverable(NodeSet nodes);
 
+    void setTotalNumberOfPendingTasks(int totalNumberOfPendingTasks);
```
Comment (amouhoub) [inline rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/ResourceManager.java]: Same thing maybe, besides it's shorter than saying NumberOf

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java [hunk 7]
```diff
@@ -2984,6 +2984,11 @@ public boolean areNodesRecoverable(NodeSet nodes) {
         return true;
     }
 
+    @Override
+    public void setTotalNumberOfPendingTasks(int totalNumberOfPendingTasks) {
```
Comment (amouhoub) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: Same as above

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/frontend/RMMonitoringImpl.java [hunk 8]
```diff
@@ -418,6 +419,11 @@ public void nodeEvent(RMNodeEvent event) {
         queueEvent(event);
     }
 
+    @Override
+    public void notifyPendingTasksCount(int count) {
```
Comment (amouhoub) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/frontend/RMMonitoringImpl.java]: Here it is OK ! but needs to be coherent elsewhere !

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/jmx/mbean/RuntimeDataMBeanImpl.java [hunk 9]
```diff
@@ -83,6 +83,11 @@ public int getAvailableNodesCount() {
         return this.rmStatisticsHolder.getStatistics().getAvailableNodesCount();
     }
 
+    @Chronological
+    public int getNumberPendingTasks() {
```
Comment (amouhoub) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/jmx/mbean/RuntimeDataMBeanImpl.java]: Same as above

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/jmx/mbean/RuntimeDataMBean.java [hunk 10]
```diff
@@ -68,6 +68,8 @@
      */
     int getAvailableNodesCount();
 
+    int getNumberPendingTasks();
```
Comment (amouhoub) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/jmx/mbean/RuntimeDataMBean.java]: Same as above

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxy.java [hunk 11]
```diff
@@ -177,4 +177,13 @@ public boolean areNodesRecoverable(NodeSet nodes) {
             return false;
         }
     }
+
+    public boolean setTotalNumberOfPendingTasks(int totalNumberOfPendingTasks) {
```
Comment (amouhoub) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxy.java]: Same as above

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxyActiveObject.java [hunk 12]
```diff
@@ -329,4 +329,9 @@ public boolean areNodesRecoverable(NodeSet nodes) {
         return rm.areNodesRecoverable(nodes);
     }
 
+    @ImmediateService
+    public boolean setTotalNumberOfPendingTasks(int totalNumberOfPendingTasks) {
```
Comment (amouhoub) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxyActiveObject.java]: Same as above

File: common/common-db/src/main/java/org/ow2/proactive/db/RrdDbUtil.java (hunk 3 above)
Comment (Mykhailenko) [reply common/common-db/src/main/java/org/ow2/proactive/db/RrdDbUtil.java]: Good thanks

File: rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/rm/RmStatsCommand.java (hunk 4 above)
Comment (Mykhailenko) [reply rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/rm/RmStatsCommand.java]: Thanks
